src/components/hooks.ts

```diff
@@ -14,3 +22,16 @@ export const useEntityDetailsParams = () => {
     userName,
   };
 };
+
+export const useModelByUUIDDetails = ({
```

Comment:
I thought it might be a good idea to create this custom hook, as we would need to get the `modelName` and `userName` in quite a few places where we want to map tree data to links. I still have to modify `ModelDetailsLink` to use it, as I didn't know if creating this custom hook would be a good approach to begin with.